openedx_learning/apps/authoring/outline_roots/models.py

```diff
@@ -0,0 +1,59 @@
+"""
+Models that implement the "outline root" for each course
+"""
+from django.db import models
+
+from ..publishing.models import Container, ContainerVersion
+
+__all__ = [
+    "OutlineRoot",
+    "OutlineRootVersion",
+]
+
+
+class OutlineRoot(Container):
+    """
+    A OutlineRoot is type of Container that defines the root of each course.
+
+    Every course run has one OutlineRoot, and it typically has a list of
+    Sections that comprise the course, which in turn have Subsections, Units,
+    and Components. However, we also allow OutlineRoot to have Subsections or
+    Units as its children, to facilitate smaller courses that don't need a
+    three-level hierarchy.
+
+    The requirements for OutlineRoot are:
+    - One OutlineRoot per course run
+    - Children must all be containers (Sections, Subsections, or Units) and
+      all children must be the same type
```

Comment:
Does it add much complexity to allow children to be a mix of different types, since we're going to have to handle different levels of traversal anyway?

Reply:
No, if anything it simplifies things because I can remove the validation checks that assert otherwise. I just kind of assumed we didn't want that, but maybe that was a bad assumption.

Reply:
Maybe let’s leave it unrestricted for now and see if there’s a pressing reason to add it later?

Reply:
OK, done.